contracts/finance/ProxiedPaymentSplitter.sol

```diff
@@ -0,0 +1,233 @@
+// SPDX-License-Identifier: MIT
+// Modified from OpenZeppelin Contracts v4.4.1 (finance/PaymentSplitter.sol)
+// Modifications copyright (c) 2021 the ethier authors (github.com/divergencetech/ethier)
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+
+/**
+ * @title ProxiedPaymentSplitter
+ * @notice This contract is almost identical to OpenZeppelin's PaymentSplitter (v4.4.1) except for the constructor
+ * being converted into an init function. This allows a single primary instance to be deployed and minimal proxy
+ * contracts used for cheaper secondary deployments.
+ *
+ * NOTE: There is no need to use this contract directly; instead see ethier's PaymentSplitterDeployer as it ensures
+ * that necessary invariants are honoured.
+ * @dev This contract allows to split Ether payments among a group of accounts. The sender does not need to be aware
+ * that the Ether will be split in this way, since it is handled transparently by the contract.
+ *
+ * The split can be in equal parts or in any other arbitrary proportion. The way this is specified is by assigning each
+ * account to a number of shares. Of all the Ether that this contract receives, each account will then be able to claim
+ * an amount proportional to the percentage of total shares they were assigned.
+ *
+ * `PaymentSplitter` follows a _pull payment_ model. This means that payments are not automatically forwarded to the
+ * accounts but kept in this contract, and the actual transfer is triggered as a separate step by calling the {release}
+ * function.
+ *
+ * NOTE: This contract assumes that ERC20 tokens will behave similarly to native tokens (Ether). Rebasing tokens, and
+ * tokens that apply fees during transfers, are likely to not be supported as expected. If in doubt, we encourage you
+ * to run tests before sending real value to this contract.
+ */
+contract ProxiedPaymentSplitter is Context {
```

Comment:
Implementing/maintaining this contract ourselves is not necessary. Openzepplin already provides upgradeable versions of their contracts (i.e. with an `initialize` function instead of a `constructor`), which I would recommend using in this case. They can be found under https://github.com/OpenZeppelin/openzeppelin-contracts  (npm package `@openzeppelin/contracts-upgradeable`)

The current implementation of the `ProxiedPaymentSplitter` contract can be replaced by the following lines, which should work out of the box.
```solidity
import "@openzeppelin/contracts-upgradeable/finance/PaymentSplitterUpgradeable.sol";

contract ProxiedPaymentSplitter is PaymentSplitterUpgradeable {
    function init(address[] memory payees, uint256[] memory shares)
        public
        initializer
    {
        __PaymentSplitter_init(payees, shares);
    }
}
```
To be consistent with other realizations of this pattern (e.g. https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/master/contracts/token/ERC20/presets/ERC20PresetFixedSupplyUpgradeable.sol), I suggest renaming `init -> initialize`.

nitpick + subjective: Ideally, I'd rename the contract `ProxiedPaymentSplitter -> PaymentSplitterUpgradeable` to be consistent with the OZ naming scheme. However, this will clash with their contract. Instead, I suggest `StandardPaymentSplitterUpgradeable`, `InitializablePaymentSplitter` or the infamous `PaymentSplitterUpgradeable_`

Reply:
Awesome, I had no idea that these existed, thank you! I'll take a closer look and make the necessary changes.

Reply:
Side note: the concept of "maintaining" is interesting here because technically this should only ever be deployed once. On that thought, should we move all such contracts into a special directory to suggest to ethier users that they don't need to actually use them in their code?

Reply:
Technically yes. But in the case that someone finds a flaw in the OZ PaymentSplitter contract at some point (admittedly very unlikely), we would have to manually propagate the changes to our implementation (and redeploy). 
Yes, I'm also very much in favour of moving this to a separate directory because they should not be considered part of the library code. Or maybe move them to a completely separate repository that builds on top of ethier?

Reply:
I've created `/factories` to house these and added comments to both re their direct use not being necessary; the entirely different directory should give developers sufficient pause to not use them.

Thanks for teaching me about the `-upgradeable` equivalents!

Reply:
I think that's a good solution.